src/providers/hana/qgshanaresultset.cpp

```diff
@@ -0,0 +1,142 @@
+/***************************************************************************
+   qgshanaresultset.h
+   --------------------------------------
+   Date      : 31-05-2019
+   Copyright : (C) SAP SE
+   Author    : Maxim Rylov
+ ***************************************************************************/
+
+/***************************************************************************
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ ***************************************************************************/
+#include "qgshanaresultset.h"
+#include "qgshanautils.h"
+#include "qgslogger.h"
+#include <QString>
+
+QgsHanaResultSet::QgsHanaResultSet( ResultSetRef &&resultSet )
+  : mResultSet( resultSet )
+  , mMetadata( mResultSet->getMetaData() )
+{
+}
+
+QgsHanaResultSetRef QgsHanaResultSet::create( StatementRef &stmt, const QString &sql )
+{
+  QgsHanaResultSetRef ret( new QgsHanaResultSet( stmt->executeQuery( reinterpret_cast<const char16_t *>( sql.unicode() ) ) ) );
+  return ret;
+}
+
+QgsHanaResultSetRef QgsHanaResultSet::create( PreparedStatementRef &stmt )
+{
+  QgsHanaResultSetRef ret( new QgsHanaResultSet( stmt->executeQuery() ) );
+  return ret;
+}
+
+void QgsHanaResultSet::close()
+{
+  mResultSet->close();
+}
+
+bool QgsHanaResultSet::next()
+{
+  return mResultSet->next();
+}
+
+QVariant QgsHanaResultSet::getValue( unsigned short columnIndex )
+{
+  switch ( mMetadata->getColumnType( columnIndex ) )
+  {
+    case SQLDataTypes::Bit:
+    case SQLDataTypes::Boolean:
+      return QgsHanaUtils::toVariant( mResultSet->getBoolean( columnIndex ), QVariant::Bool );
+    case SQLDataTypes::Char:
+      return QgsHanaUtils::toVariant( mResultSet->getByte( columnIndex ) );
```

Comment:
`SQLDataTypes::Char` does not mean that it's a single character. It just means that the column has a fixed width, e.g. `CREATE TABLE TAB(ID CHAR(8))`.

Reply:
Done in 4d36f6f.